### PR TITLE
Improve OAI response speed, refs #9608 and #9618

### PR DESCRIPTION
**Pull request by jraddaoui**
- Avoid unnecessary calls to QubitInformationobject::getCollections().
- Filter drafts instead of ACL checks in that function.
- Filter drafts in QubitInformationobject::getUpdatedRecords().
- Check ACL only in the ListRecords and GetRecord verb templates.
- Other small fixes and cosmetic changes.

### CODE REVIEW

**Comment by sevein** [comment]
Is this ready for CR?

**Comment by jraddaoui** [comment]
https://projects.artefactual.com/issues/9608#note-5

**Comment by sevein** [comment]
CC @mcantelon

**Comment by sevein** [comment]
@jraddaoui thank you for the multiple meaningful commits, very useful.

File: plugins/arOaiPlugin/lib/arOaiPluginComponent.class.php [hunk 1]
```diff
@@ -96,7 +96,7 @@ public function setUpdateParametersFromRequest($request)
     }
   }
 
-  public function getUpdates()
+  public function getUpdates($filterDrafts = false)
```
Comment (sevein) [inline plugins/arOaiPlugin/lib/arOaiPluginComponent.class.php]: Can you use the `array $options` pattern? I'm not doing CR of this pull request but I ran into this accidentally.

File: lib/model/QubitInformationObject.php [hunk 2]
```diff
@@ -321,42 +321,54 @@ public function delete($connection = null)
...
 
-    if ($until != '')
+    if (!empty($options['set']))
     {
-      $criteria->addAnd(QubitObject::UPDATED_AT, $until, Criteria::LESS_EQUAL);
+      $options['set']->apply($criteria);
     }
 
-    if ($set != '')
+    if (isset($options['filterDrafts']) && $options['filterDrafts'])
```
Comment (mcantelon) [inline lib/model/QubitInformationObject.php:349]: Nitpick: could use `!empty($options['filterDrafts'])`

File: plugins/arOaiPlugin/modules/arOaiPlugin/actions/indexAction.class.php [hunk 3]
```diff
@@ -48,7 +48,7 @@ public function execute($request)
   {
     sfConfig::set('sf_escaping_strategy', false);
 
-    $appRoot = dirname(__FILE__) .'/../../../../..';
+    $appRoot = sfConfig::get('sf_root_dir');
```
Comment (mcantelon) [inline plugins/arOaiPlugin/modules/arOaiPlugin/actions/indexAction.class.php:51]: Nice!

**Comment by mcantelon** [comment]
Great optimization and cleanup! Looks good!

File: lib/model/QubitInformationObject.php (hunk 2 above)
Comment (jraddaoui) [inline lib/model/QubitInformationObject.php:349]: Hi Mike, in this case it's a boolean and it could be not empty but false.

**Pull request closed without merging**